plugin/forward/forward.go

```diff
@@ -65,24 +66,36 @@ func (f *Forward) ServeDNS(ctx context.Context, w dns.ResponseWriter, r *dns.Msg
 	}
 
 	fails := 0
-	var span, child ot.Span
-	var upstreamErr error
-	span = ot.SpanFromContext(ctx)
+	i := 0
+	returnErr := errNoHealthy
+	span := ot.SpanFromContext(ctx)
```

Comment:
why did you move the `child` to down below?

Reply:
1. To reduce the scope of `child`. It is not referred outside the loop
2. This way `child` is initialized with `nil` in each iteration. Theoretically `child` may not be assigned with value in current iteration, and later down the code we compare it with `nil` to call `Finish()`. Non-nil value of `child` from previous iteration could lead to repeated call to `Finish()` which seems incorrect. Though, I guess such case is unlikely on practice